src/trackedit/internal/au3/au3selectioncontroller.cpp

```diff
@@ -101,42 +95,7 @@ bool Au3SelectionController::isDataSelected() const
 
 bool Au3SelectionController::isDataSelectedOnTrack(TrackId trackId) const
```

Comment:
About `isDataSelected()` : I'm not sure what it is supposed to tell its clients? For sure it's only about time (it doesn't consider whether a track is selected or not). It would return true if start time were 10 and end time were 1.
Start and end time setters are part of the public interface, and I don't see sanity checks to make sure that end time is never set to something less than start time, so it could happen. I'm not saying that these checks should be added now, but if this isn't intended to be possible and `isDataSelected` intends in fact to say `timeSelectionIsNotEmpty`, then I would like it to be renamed as such. It is used in many places in the code, and this could reveal bugs.

Likewise for `isDataSelectedOnTrack`: what it tells now is only whether the track of the given ID is selected or not, so I would like this see renamed to `isTrackSelected`.

Reply:
I forgot: the implementation of `isDataSelected` should also add the `&&` that end time is strictly larger than start time (if negative selection spans are indeed not intended).